src/server/ua_subscription_alarms_conditions.c

```diff
@@ -0,0 +1,2576 @@
+                                const UA_NodeId *sessionId, void *sessionContext,
+                                const UA_NodeId *nodeId, void *nodeContext,
+                                const UA_NumericRange *range, const UA_DataValue *data) {
+    //TODO
+}
+
+static void
+afterWriteCallbackSeverityChange(UA_Server *server,
+                                 const UA_NodeId *sessionId, void *sessionContext,
+                                 const UA_NodeId *nodeId, void *nodeContext,
+                                 const UA_NumericRange *range, const UA_DataValue *data) {
+    UA_QualifiedName fieldLastSeverity = UA_QUALIFIEDNAME(0, CONDITION_FIELD_LASTSEVERITY);
+    UA_QualifiedName fieldSourceTimeStamp = UA_QUALIFIEDNAME(0, CONDITION_FIELD_CONDITIONVARIABLE_SOURCETIMESTAMP);
+    UA_QualifiedName fieldMessage = UA_QUALIFIEDNAME(0, CONDITION_FIELD_MESSAGE);
+    UA_QualifiedName fieldTime = UA_QUALIFIEDNAME(0, CONDITION_FIELD_TIME);
+    UA_NodeId condition;
+    UA_LocalizedText message;
+    UA_Variant value;
+
+    UA_StatusCode retval = getFieldParentNodeId(server, nodeId, &condition);
+    CONDITION_ASSERT_RETURN_VOID(retval, "No Parent Condition found for given Severity Field",);
+
+    /* Get conditionSource */
+    UA_NodeId conditionSource;
+    retval = getNodeIdValueOfConditionField(server, &condition, UA_QUALIFIEDNAME(0, CONDITION_FIELD_SOURCENODE), &conditionSource);
+    CONDITION_ASSERT_RETURN_VOID(retval, "ConditionSource not found", UA_NodeId_deleteMembers(&condition););
+
+    UA_LastSverity_Data outLastSeverity;
+    retval = getConditionLastSeverity(server, &conditionSource, &condition, &outLastSeverity);
+    CONDITION_ASSERT_RETURN_VOID(retval, "Get Condition LastSeverity failed", UA_NodeId_deleteMembers(&condition);
+                                 UA_NodeId_deleteMembers(&conditionSource););
+
+    /* Set message dependent on compare result*/
+    if(outLastSeverity.lastSeverity < (*(UA_UInt16 *)data->value.data))
+        message = UA_LOCALIZEDTEXT(LOCALE, SEVERITY_INCREASED_MESSAGE);
+    else
+        message = UA_LOCALIZEDTEXT(LOCALE, SEVERITY_DECREASED_MESSAGE);
+
+    /* Set LastSeverity */
+    UA_Variant_setScalar(&value, &outLastSeverity.lastSeverity, &UA_TYPES[UA_TYPES_UINT16]);
+    retval = UA_Server_setConditionField(server, condition, &value, fieldLastSeverity);
+    CONDITION_ASSERT_RETURN_VOID(retval, "Set Condition LAstSeverity failed", UA_NodeId_deleteMembers(&condition);
+                                 UA_NodeId_deleteMembers(&conditionSource););
+
+    /* Set SourceTimestamp */
+    UA_Variant_setScalar(&value, &outLastSeverity.sourceTimeStamp, &UA_TYPES[UA_TYPES_DATETIME]);
+    retval = UA_Server_setConditionVariableFieldProperty(server, condition, &value, fieldLastSeverity, fieldSourceTimeStamp);
+    CONDITION_ASSERT_RETURN_VOID(retval, "Set LastSeverity SourceTimestamp failed", UA_NodeId_deleteMembers(&condition);
+                      
```

Comment:
Not needed anymore, because it is allowed now to trigger Events of abstract EventTypes.